thirdparty/lua-htmlparser/CMakeLists.txt

```diff
@@ -0,0 +1,46 @@
+project(lua-htmlparser)
+cmake_minimum_required(VERSION 3.5.1)
+
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_LIST_DIR}/../cmake_modules")
+include("koreader_thirdparty_common")
+include("koreader_thirdparty_git")
+
+enable_language(C)
+
+assert_var_defined(CC)
+assert_var_defined(CFLAGS)
+assert_var_defined(LD)
+assert_var_defined(LUA_INCDIR)
+assert_var_defined(LUA_LIBDIR)
+assert_var_defined(LUA_HTMLPARSER_VER)
+assert_var_defined(OUTPUT_DIR)
+assert_var_defined(LUAROCKS)
+
+ep_get_source_dir(SOURCE_DIR)
+
+set(LUA_HTMLPARSER_ROCKSPEC rockspec/${LUA_HTMLPARSER_VER}.rockspec)
+
+set(BUILD_CMD sh -c "${LUAROCKS} make --tree=${OUTPUT_DIR}/rocks ${LUA_HTMLPAPRSER_ROCKSPEC}")
+if(DEFINED LIBFLAG)
+    set(BUILD_CMD "${BUILD_CMD} LIBFLAG=\"${LIBFLAG}\"")
+endif()
+set(BUILD_CMD "${BUILD_CMD} CC=\"${CC}\" CFLAGS=\"${CFLAGS}\" LD=\"${LD}\"")
+set(BUILD_CMD "${BUILD_CMD} LUA_INCDIR=\"${LUA_INCDIR}\" LUA_LIBDIR=\"${LUA_LIBDIR}\"")
+
+ko_write_gitclone_script(
+    GIT_CLONE_SCRIPT_FILENAME
+    https://github.com/msva/lua-htmlparser
+    tags/v0.3.6
```

Comment:
`${LUA_HTMLPARSER_VER}`?

Reply:
Will this work, lua-Spore uses this way?

Reply:
Spore doesn't have `v0.3.6` in both. But actually you pointed out something I hadn't noticed, namely that `v0.3.6` in `LUA_HTMLPARSER_VER` isn't a value that makes sense.

We can probably go for commit [4f6437ebd123c3e552a595fc818fdd952888fff2](https://github.com/msva/lua-htmlparser/commit/4f6437ebd123c3e552a595fc818fdd952888fff2) here, since it looks like a useful fix, which would give you `scm-0` for `LUA_HTMLPARSER_VER`.

Reply:
Will this work with rockspec?

Reply:
v0.3.6 *doesn't* work, it already falls back to scm-0.

Reply:
The correct value would be `0.3.6-1`, except that's not part of the tag and I want the more recent bugfix regardless.